### PR TITLE
setup.md file

### PR DESCRIPTION
**Pull request by ledm**
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [ ] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [ ] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [ ] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes {#45}

### CODE REVIEW

File: setup.md [hunk 1]
```diff
@@ -1,7 +1,157 @@
...
+
+### DLR 
+FIXME
+
+### Other computing systems
 FIXME
 
+### Your own machine
+
+If you are planning on running ESMValTool on your own machine, please make sure that you are able to download CMIP data and that you have several GB of space available to install conda & ESMVAlTool, but also enough to make a copy of some data.
```
Comment (sverhoeven) [inline setup.md:107]: For someone without access to an ESGF node it would be nice if he/she could run a plot recipy without having to know which data he/she needs to download before. As ESMValTool can use [synda](http://prodiguer.github.io/synda/index.html) to download CMIP data. 

It would be nice to install synda in this section. Something like instructions at https://docs.esmvaltool.org/en/latest/input.html#models

File: setup.md [hunk 2]
```diff
@@ -1,7 +1,157 @@
...
+
+Here’s an explanation of the github issues: https://guides.github.com/features/issues/
+
+Raising an issue is the act of creating a new issue. If you are asked to raise an issue, please follow any instructions that you are given, and also make sure that you read the default issue text. 
+
+### pull request 
+
+A github pull request is the act of requesting that a branch is merged with another branch.
+
+This is an advanced feature of ESMValTool, and will generally be performed by the Core development team.
```
Comment (sverhoeven) [inline setup.md:143]:
```suggestion
This is an advanced feature of GitHub, and will generally be performed by the ESMValTool  development team. 
```

File: setup.md [hunk 3]
```diff
@@ -1,7 +1,157 @@
...
+This is an advanced feature of ESMValTool, and will generally be performed by the Core development team. 
+
+
+
+
+## Install conda
+
+The python package manager Conda (anaconda or miniconda) needs to be installed on your system before the tutorial starts. In some cases, your system may have a central version installed already. 
+
+More details on this process are available in the Installation episode.
```
Comment (sverhoeven) [inline setup.md:152]: Link it up
```suggestion
More details on this process are available in the [Installation episode]({{ page.root}}[% link _episodes/02-installation.md  %}). 
```

File: setup.md [hunk 4]
```diff
@@ -1,7 +1,157 @@
...
+
+#### Jasmin-login
+
+Note that you have only created an account for the web-interface. To log into the jasmin machine and do work, you&#39;ll need to create a login account too: https://help.jasmin.ac.uk/article/161-get-login-account 
+
+#### Access to data on JASMIN
+
+Please request access to the working groups:
+● https://accounts.jasmin.ac.uk/services/group_workspaces/esmeval 
+●https://services.ceda.ac.uk/cedasite/resreg/application?attributeid=cmip5_research
```
Comment (sverhoeven) [inline setup.md:74]: `●` does not render a bullet list use `-`.

```suggestion
- https://accounts.jasmin.ac.uk/services/group_workspaces/esmeval 
- https://services.ceda.ac.uk/cedasite/resreg/application?attributeid=cmip5_research 
```

File: setup.md [hunk 5]
```diff
@@ -1,7 +1,157 @@
 ---
-title: Setup
+title: Preparing for participating in the tutorial 
 ---
+
+This page includes some basic information on how to prepare to participate in this tutorial. 
+
+> ## Prerequisites
+> The prerequisites for the tutorial are listed here: 
+> Basic understanding of git (optional - but useful)
```
Comment (ledm) [inline setup.md:9]: this list doesn't display properly.

File: setup.md [hunk 6]
```diff
@@ -1,7 +1,157 @@
...
+
+
+## Optional tutorials
+
+Where available, we include links to other software carpentry courses. 
+
+### Command line
+
+We typically use the command line to interact with ESMValTool. Here’s a software carpentry tutorial on the unix shell:
+https://swcarpentry.github.io/shell-novice/
```
Comment (ledm) [inline setup.md:24]: link doesn't work

File: setup.md [hunk 7]
```diff
@@ -1,7 +1,157 @@
...
+
+
+### Basic understanding of git
+ 
+Git is a distributed version-control system for tracking changes in source code during software development. It’s how we distribute, share, and manage the ESMValTool code. 
+
+There are many basic introductions to git available.
+ 
+Here’s a software carpentry tutorial on git:
+https://swcarpentry.github.io/git-novice/
```
Comment (ledm) [inline setup.md:34]: bad link

File: setup.md [hunk 8]
```diff
@@ -1,7 +1,184 @@
...
+
+If neccesairy, data can be downloaded using the 
+[synda tool](https://prodiguer.github.io/synda/index.html).
+ 
+
+
+### CEDA-Jasmin
+
+If you do not already have an account on JASMIN, then request an account as
+soon as possible. The instructions on how to create a Jasmin account are here: https://help.jasmin.ac.uk/article/4435-get-a-jasmin-account-portal
```
Comment (sverhoeven) [inline setup.md:81]: Bare urls don't get converted to hyperlinks
```suggestion
soon as possible. The instructions on how to create a Jasmin account are [here]( https://help.jasmin.ac.uk/article/4435-get-a-jasmin-account-portal)
```

File: setup.md [hunk 9]
```diff
@@ -1,7 +1,184 @@
...
+
+During the account creation, you will need an SSH key, which can be generated following these instructions: https://help.jasmin.ac.uk/article/185-generate-ssh-key-pair
```
Comment (sverhoeven) [inline setup.md:83]:
```suggestion
During the account creation, you will need an SSH key, which can be generated following [these instructions](https://help.jasmin.ac.uk/article/185-generate-ssh-key-pair)
```

File: setup.md [hunk 10]
```diff
@@ -1,7 +1,184 @@
...
+
+### CEDA-Jasmin
+
+If you do not already have an account on JASMIN, then request an account as
+soon as possible. The instructions on how to create a Jasmin account are here: https://help.jasmin.ac.uk/article/4435-get-a-jasmin-account-portal
+
+During the account creation, you will need an SSH key, which can be generated following these instructions: https://help.jasmin.ac.uk/article/185-generate-ssh-key-pair 
+
+More instructions on how to get started:
+https://help.jasmin.ac.uk/article/189-get-started-with-jasmin
```
Comment (sverhoeven) [inline setup.md:86]:
```suggestion
[https://help.jasmin.ac.uk/article/189-get-started-with-jasmin](https://help.jasmin.ac.uk/article/189-get-started-with-jasmin)
```

File: setup.md [hunk 11]
```diff
@@ -1,7 +1,216 @@
 ---
-title: Setup
+title: Preparations for participating in the tutorial 
 ---
+
+This page includes some basic information on how to prepare to participate in this tutorial.
```
Comment (ledm) [inline setup.md:5]:
```suggestion
This page includes some information on how to prepare for participating in this tutorial. 
```

File: setup.md [hunk 12]
```diff
@@ -1,7 +1,216 @@
...
+>
+> - Access to CMIP data
+>
+> - Access to a suitable computing system (eg Jasmin, DLR machine)
+>
+> - Github account (optional)
+{: .prereq}
+
+
+## Optional tutorials
```
Comment (ledm) [inline setup.md:23]:
```suggestion
## Optional tutorials to be prepared to use ESMValTool
```

File: setup.md [hunk 13]
```diff
@@ -1,7 +1,212 @@
...
+
+
+## Github account (Advanced)
+
+You don’t need a github account to participate in the tutorial. 
+However, if you want to raise an issue, contribute to the discussions, 
+or share your code, please [create a github account](https://github.com/).
+
+To learn how to use github, please have a look at this [introduction to github]
+(https://lab.github.com/githubtraining/introduction-to-github).
```
Comment (sverhoeven) [inline setup.md:176]: Can not have link on split over 2 lines
```suggestion
To learn how to use github, please have a look at this [introduction to github](https://lab.github.com/githubtraining/introduction-to-github).
```

File: setup.md [hunk 14]
```diff
@@ -1,7 +1,296 @@
...
+~~~
+{: .language-bash}
+
+Note that the JASMIN is only open to certain locations (mostly universities, and research centres). 
+You may need a VPN if you wish to connect from your home network.
+
+
+### DKRZ
+
+If you do not already have an account at the DKRZ, then [register](https://luv.dkrz.de/projects/newuser/) as soon as possible. You could find a short introduction how to get started at DKRZ [here](https://www.dkrz.de/up/my-dkrz/getting-started/getting-started-at-dkrz).
```
Comment (hb326) [inline setup.md:187]: I would rephrase here to "You can find a short introduction on how to get started at DKRZ..."

File: setup.md [hunk 15]
```diff
@@ -1,7 +1,296 @@
...
+
+There is also an [user manual](https://www.dkrz.de/up/systems/mistral) for mistral which is DKRZ's current supercomputer.
```
Comment (hb326) [inline setup.md:189]: Maybe rephrase to "There is also a [user manual]..."
And exchange "mistral" with "Mistral".

File: setup.md [hunk 16]
```diff
@@ -1,7 +1,296 @@
...
+
+#### Join a project
+
+To use the resources on DKRZ you have to join a project. You could join an existing project by logging into [https://luv.dkrz.de/](https://luv.dkrz.de/) with your account and select 'Join existing project'. Once you are accepted by a project we will turn your web account into a full LDAP account which allows you to log into and use our systems. If you have no access to an existing project here are some instructions [how to apply for resources](https://www.dkrz.de/services/bereitstellung-von-rechenleistung?set_language=en&cl=en).
+
+#### Access to data on DKRZ
+
+Availability of CMIP5 and CMIP6 data in this directories:
```
Comment (hb326) [inline setup.md:197]: I would rephrase: "CMIP5 and CMIP6 data are available in these directories:"

File: setup.md [hunk 17]
```diff
@@ -1,7 +1,296 @@
...
+
+### DKRZ
+
+If you do not already have an account at the DKRZ, then [register](https://luv.dkrz.de/projects/newuser/) as soon as possible. You could find a short introduction how to get started at DKRZ [here](https://www.dkrz.de/up/my-dkrz/getting-started/getting-started-at-dkrz).
+
+There is also an [user manual](https://www.dkrz.de/up/systems/mistral) for mistral which is DKRZ's current supercomputer.
+
+#### Join a project
+
+To use the resources on DKRZ you have to join a project. You could join an existing project by logging into [https://luv.dkrz.de/](https://luv.dkrz.de/) with your account and select 'Join existing project'. Once you are accepted by a project we will turn your web account into a full LDAP account which allows you to log into and use our systems. If you have no access to an existing project here are some instructions [how to apply for resources](https://www.dkrz.de/services/bereitstellung-von-rechenleistung?set_language=en&cl=en).
```
Comment (hb326) [inline setup.md:193]: I would rephrase "One option is to join an existing project ... with your account information .... Once you are accepted by the manager of your chosen project, your web account will be turned into a full LDAP account which will allow you to log into and use the DKRZ's resources."
Maybe also rephrase this sentence a little "If you do not have access to an existing project, another option for you would be to apply for resources at DKRZ. Here are some instructions on [how to apply for resources]..."

File: setup.md [hunk 18]
```diff
@@ -1,7 +1,296 @@
...
+(see also [this](https://www.dkrz.de/up/systems/mistral/hpc-concepts))
+
+Data storage:
+- Personal data: home directory (24GiB)
+- Project data: /work/<project>/<user-account>
+- Temporary data: scratch directory on /scratch/*/<user-account> will be automatically deleted after 14 days (15TiB) (Please use this directory for tests! Do not use the work directory for tests.)
+(see also [this](https://www.dkrz.de/up/systems/mistral/hpc-concepts))
+
+Running batch jobs:
+Info and examples on SLURM job scheduling system at DKRZ could be found [here](https://www.dkrz.de/up/systems/mistral/running-job).
```
Comment (hb326) [inline setup.md:225]: I probably would rephrase to "... at DKRZ can be found [here]..."

File: setup.md [hunk 19]
```diff
@@ -1,7 +1,296 @@
...
+Login nodes are for compiling and job submission only! For all other tasks, use one of the pre- and post-processing nodes
+~~~
+ssh -X <user-account>@mistralpp.dkrz.de
+~~~
+(see also [this](https://www.dkrz.de/up/systems/mistral/hpc-concepts))
+
+Data storage:
+- Personal data: home directory (24GiB)
+- Project data: /work/<project>/<user-account>
+- Temporary data: scratch directory on /scratch/*/<user-account> will be automatically deleted after 14 days (15TiB) (Please use this directory for tests! Do not use the work directory for tests.)
```
Comment (hb326) [inline setup.md:221]: I would probably rephrase "Temporary data: the scratch directory on /scratch/*/<user-account> is automatically... (Please use this directory for all your testing! ..."

**Comment by hb326** [comment]
@LisaBock: I added some comments to the text. 
@ledm: @LisaBock has written the setup description for DKRZ, and I have reviewed it just now.

**Comment by LisaBock** [comment]
> 
> 
> @LisaBock: I added some comments to the text.
> @ledm: @LisaBock has written the setup description for DKRZ, and I have reviewed it just now.

Thanks @hb326 ! I added all your suggestions.
@ledm The DKRZ part is now finished.

File: setup.md (hunk 1 above)
Comment (ledm) [reply setup.md:233]: I've added thr synda stuff to the required packages, but it probably needs a lot more detail. In fact, the whole "your own machine" could use a bit of work.

File: setup.md [hunk 20]
```diff
@@ -1,7 +1,157 @@
...
+There are many basic introductions to git available.
+ 
+Here’s a software carpentry tutorial on git:
+https://swcarpentry.github.io/git-novice/
+
+
+
+## Access to CMIP and OBS data and a suitable compute cluster
+
+ESMValTool may be run on multiple platforms, including local machines and
```
[the remark this answers is not in the document]
Comment (ledm) [reply setup.md]: I think this is done now.

File: setup.md (hunk 1 above)
Comment (sverhoeven) [reply setup.md:233]: I created https://github.com/ESMValGroup/tutorial/issues/54 to improve the section later.

**Comment by ledm** [comment]
Lol, I was still working here!

**Comment by ledm** [comment]
Still, glad that it's merged. I can always make another PR with minor changes.

**Comment by hb326** [comment]
> 
> 
> Lol, I was still working here!

Ahh, sorry, Lee! I saw that Stefan had approved the pull request and so I thought I merge it. 
But I already saw that you have opened another for the minor changes.
